### PR TITLE
Deprecate FastBlock Flag

### LINKED ISSUE #6295: Deprecate `FastBlocks` flag
**Is your feature request related to a problem? Please describe.**
There are two flags in Sync config - `FastSync` and `FastBlocks`. Second one is causing confusion sometimes - nodes are not able to sync if `FastSync` is `true`, but `FastBlocks` not. I'm not familiar with any use case when one of this flags is true and the other is false.

**Describe the solution you'd like**
Keep `FastBlocks` as part of the config to not break user setups, but don't use it on code - use `FastSync` instead

**Describe alternatives you've considered**
Keep it as it is

**Additional context**
I recently spent some time to figure out the cause of not syncing with message
`Unable to find beacon header at height 1. This is unexpected, forcing a new beacon sync`
Lack of `FastBlocks` = `true` was the reason

### PR DESCRIPTION
**Pull request by obasekiosa**
Fixes: #6295 

## Changes

- Replace FastBlocks with FastSync
- Marks FastBlocks as deprecated on SyncConfig Interface
- Removes FastBlocks from config files
- Updates ParallelSync tests to accommodate for FastHeaders being synced before testing.
- Removes commented out tests

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [x] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Notes on testing

Some tests where updated to remove FastBlocks, and some parallel sync tests where updated to assume FastHeaders have already been synced before testing. 

## Documentation
Enabling fast sync now automatically enables fast blocks. Since fast blocks depends on fast sync it is no longer needed as a configuration option.

#### Requires documentation update

- [x] Yes
- [ ] No

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

Documentation  and release notes should indicate FastBlocks has been deprecated. Although it would be ignored if added, currently adding it to config would not break setups.

## Remarks

During a set of tests by @marcindsobczak it was noticed that some Syncing tests faild with the error message `Unable to find beacon header at height 1. This is unexpected, forcing a new beacon sync`. 
The reason for these errors were FastBlocks being false when FastSync was set to true. Given that that case is hardly ever if ever at all used. It was decided its better to remove FastBlocks completed, deprecating it and replacing its use in logic to the value of FastSync. 

This fix is the first phase of fully deprecating FastBlocks

### CODE REVIEW

**Review by MarekM25 — CHANGES_REQUESTED** [review]
This wasn't the objective of this issue. Let's settle the internal discussion with @LukaszRozmej, and then we can schedule a call to provide you with a better understanding.

**Comment by obasekiosa** [comment]
ignore this, I wanted to see all tests that would fail in the case if fastsync was forced.
Can't do that locally due to system dependent flaky tests.

File: src/Nethermind/Nethermind.Synchronization.Test/FastBlocks/FastBlocksFeedTests.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@
 //             _syncConfig.PivotNumber = _validTree2048.Head.Number.ToString();
 //             _syncConfig.PivotTotalDifficulty = _validTree2048.Head.TotalDifficulty.ToString();
 //             _syncConfig.UseGethLimitsInFastBlocks = false;
-//             _syncConfig.FastBlocks = true;
+//             _syncConfig.FastSync = true;
```
[the remark this answers is not in the document]
Comment (obasekiosa) [reply src/Nethermind/Nethermind.Synchronization.Test/FastBlocks/FastBlocksFeedTests.cs]: I had thought they were used to generate some kind of tests.

File: src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs [hunk 2]
```diff
@@ -83,32 +83,18 @@ public long FindBestFullState()
...
         public bool IsSnapGetRangesFinished() => _snapSyncFeed?.IsFinished ?? true;
 
         public void RecalculateProgressPointers() => _blockTree.RecalculateTreeLevels();
 
-        private bool IsFastBlocks()
-        {
-            // maybe eliminate the entire code refeferencing this function why? fast block depends on fastsync so this code might just be check that is now unnecessay!
-            // name isFastBlocks left in place for correlation with protocol specification.
-            bool isFastBlocks = _syncConfig.FastSync; 
+        private bool IsFastBlocks() => _syncConfig.FastSync && _syncConfig.PivotNumberParsed != 0L; // if pivot number is 0 then it is equivalent to fast blocks disabled
```
Comment (obasekiosa) [inline src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs:96]: @LukaszRozmej 
Why is does this line work.
Seems to be the only change in logic within the fixes

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs:96]: its just logic simplification - works exactly the same

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Synchronization.Test/FastBlocks/FastBlocksFeedTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization.Test/FastBlocks/FastBlocksFeedTests.cs]: Looks like someone left those obsolete tests - deleted them

**Comment by obasekiosa** [comment]
I see - the test needed to change

File: src/Nethermind/Nethermind.Blockchain/Synchronization/SyncConfig.cs [hunk 3]
```diff
@@ -13,10 +12,10 @@ public class SyncConfig : ISyncConfig
         private bool _fastSync;
 
         public static ISyncConfig Default { get; } = new SyncConfig();
-        public static ISyncConfig WithFullSyncOnly { get; } = new SyncConfig { FastSync = false, FastBlocks = false };
+        public static ISyncConfig WithFullSyncOnly { get; } = new SyncConfig { FastSync = false };
         public static ISyncConfig WithFastSync { get; } = new SyncConfig { FastSync = true };
-        public static ISyncConfig WithFastBlocks { get; } = new SyncConfig { FastSync = true, FastBlocks = true };
-        public static ISyncConfig WithEth2Merge { get; } = new SyncConfig { FastSync = false, FastBlocks = false, BlockGossipEnabled = false };
+        public static ISyncConfig WithFastBlocks { get; } = new SyncConfig { FastSync = true };
```
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Blockchain/Synchronization/SyncConfig.cs:17]: Just delete this whole line

**Review by MarekM25 — APPROVED** [review]
@obasekiosa Have you synced one node with one-click to confirm that code works correctly? If you don't know how - let me know! :)

**Comment by obasekiosa** [comment]
> @obasekiosa Have you synced one node with one-click to confirm that code works correctly? If you don't know how - let me know! :)

I don't know how could you walk me through the overview, left a dm.

**Comment by xanatos** [comment]
Don't know if it is connected, but with 1.26 on MainNet I'm getting 

> Incorrect config settings found:
> ConfigType:JsonConfigFile|Category:SyncConfig|Name:FastBlocks

And if I check the mainnet.cfg of the 1.26 I have

> "FastBlocks": true,

**Comment by obasekiosa** [comment]
> Don't know if it is connected, but with 1.26 on MainNet I'm getting
> 
> > Incorrect config settings found:
> > ConfigType:JsonConfigFile|Category:SyncConfig|Name:FastBlocks
> 
> And if I check the mainnet.cfg of the 1.26 I have
> 
> > "FastBlocks": true,

Fast block was deprecated, but it being present in the config shouldn't raise an error.
A warning yes but not an error.
it's left there to allow for smooth transition, but it's not being used.

**Pull request merged**